assets/javascript/apps/pipeline/widgets.tsx

```diff
@@ -3,13 +3,101 @@ import React, {
   ChangeEventHandler,
   Dispatch,
   SetStateAction,
+  useId,
 } from "react";
 import { InputParam } from "./types/nodeInputTypes";
 import { NodeParameterValues } from "./types/nodeParameterValues";
 import usePipelineStore from "./stores/pipelineStore";
 import { NodeParams } from "./types/nodeParams";
 import { NodeProps } from "reactflow";
 
+export function TextModal({
+  humanName,
+  name,
+  value,
+  onChange,
+}: {
+  humanName: string;
+  name: string;
+  value: string | string[];
+  onChange: ChangeEventHandler;
+}) {
+  const modalId = useId();
+  return (
+    <>
+      <dialog
+        id={modalId}
+        className="modal nopan nodelete nodrag noflow nowheel"
+      >
+              <div className="modal-box  min-w-[85vw] h-[80vh] flex flex-col">
+          <form method="dialog">
+            <button className="btn btn-sm btn-circle btn-ghost absolute right-2 top-2">
+              ✕
+            </button>
+          </form>
+          <div className="flex-grow h-full w-full">
+            <h4 className="mb-4 font-bold text-lg bottom-2 capitalize">
+              {humanName}
+            </h4>
+            <textarea
+              className="textarea textarea-bordered textarea-lg h-[80%] w-full"
+              name={name}
+              onChange={onChange}
+              value={value}
+            ></textarea>
+            <form method="dialog" className="modal-backdrop">
+              <button className="pg-button-primary mt-2">Save</button>
+            </form>
+          </div>
+        </div>
+        <form method="dialog" className="modal-backdrop">
+          {/* Allows closing the modal by clicking outside of it */}
+          <button>close</button>
+        </form>
+      </dialog>
+      <button
+        className="btn btn-ghost"
+        onClick={() =>
+          (document.getElementById(modalId) as HTMLDialogElement)?.showModal()
+        }
+      >
+        <i className="fa-solid fa-pencil"></i>
```

Comment:
Nit: How about using either [this](https://fontawesome.com/icons/expand?s=solid) or [this](https://fontawesome.com/v5/icons/expand-alt?s=solid) icon to show that this can be expanded. For me the pencil indicates that I can edit the text, but I can do so by typing in the textbox already, so it would be a little confusing to me. Showing the expand icon will hopefully convey the message "hey user, you can expand the textbox".